torch/csrc/utils/tensor_new.cpp

```diff
@@ -206,45 +189,42 @@ Tensor internal_new_from_data(
     bool copy_variables,
     bool copy_numpy,
     bool type_inference) {
-  int32_t device_index = -1;
-  if (device_opt.has_value()) {
-    device_index = device_opt->index();
-  }
   if (THPUtils_checkString(data)) {
     throw TypeError("new(): invalid data type '%s'", Py_TYPE(data)->tp_name);
   }
 
   if (THPVariable_Check(data)) {
     auto var = reinterpret_cast<THPVariable*>(data)->cdata;
-    auto type_inference_device_type = device_opt.has_value() ? device_opt->type()
-                                                             : torch::getDeviceType(var.type());
+    if (copy_variables) {
+      var = autograd::make_variable(var.data(), /*requires_grad=*/false);
```

Comment:
why do you need to do this?  Or rather, why didn't we need to do it before?

Reply:
It's to disconnect it hard in the autograd graph when using `torch.tensor(...)`. My theory is that previously we used `Type.copy` which is autograd-breaking. Of course, one might ask whether it'd be cleaner to use `.detach()`.